doc/sphinxext/gallery_generator.py

```diff
@@ -216,25 +232,27 @@ def extract_docstring(self):
         # thumbloc = None
         # title: Optional[str] = None
         ex_title: str = ""
+        gallery_category = None
         for line in docstring.split("\n"):
-            # # we've found everything we need...
+            # we've found everything we need...
             # if thumbloc and title and ex_title != "":
             #     break
             # m = re.match(r"^_thumb: (\.\d+),\s*(\.\d+)", line)
             # if m:
-            #     thumbloc = float(m.group(1)), float(m.group(2))
-            #     continue
-            # m = re.match(r"^_example_title: (.*)$", line)
-            # if m:
-            #     title = m.group(1)
-            #     continue
+                # thumbloc = float(m.group(1)), float(m.group(2))
+                # continue
+            m = re.match(r"^_gallery_category: (.*)$", line)
+            if m:
+                gallery_category = m.group(1).title()
+                continue
             # capture the first non-empty line of the docstring as title
             if ex_title == "":
                 ex_title = line
         assert ex_title != ""
-        # if thumbloc is not None:
-        #     self.thumbloc = thumbloc
-        #     docstring = "\n".join([l for l in docstring.split("\n") if not l.startswith("_thumb")])
+        self.gallery_category = "Miscellaneous"  # Default to category-less
+        if gallery_category is not None:
+            self.gallery_category = gallery_category
+            docstring = "\n".join([l for l in docstring.split("\n") if not l.startswith("_gallery_category")])
```

Comment:
Add docstring. While a clever one liner, id make this its own method that says "get docstring" or something similar so its more clear what's going on